src/data_provider/src/extended_sources/users/src/sudoers_unix.cpp

```diff
@@ -0,0 +1,234 @@
+/* Copyright (C) 2015, Wazuh Inc.
+ * All rights reserved.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include <algorithm>
+#include <sstream>
+#include <iostream>
+#include <regex>
+#include <locale>
+#include <vector>
+#include <filesystem>
+#include <fstream>
+
+#include "stringHelper.h"
+
+#include "sudoers_unix.hpp"
+
+SudoersProvider::SudoersProvider(const std::string fileName)
+    : m_sudoFile(fileName)
+{
+}
+
+SudoersProvider::SudoersProvider()
+    : m_sudoFile("/etc/sudoers")
+{
+}
+
+nlohmann::json SudoersProvider::collect()
+{
+    nlohmann::json results = nlohmann::json::array();
+
+    genSudoersFile(m_sudoFile, 1, results);
+
+    return results;
+}
+
+
+void SudoersProvider::genSudoersFile(const std::string& fileName,
+                                     unsigned int level,
+                                     nlohmann::json& results)
+{
+    // sudoers(5): No more than 128 files are allowed to be nested.
+    static const unsigned int kMaxNest = 128;
+
+    // Local utility functions/lambdas
+    auto trim = [](std::string & s)
+    {
+        s.erase(s.begin(), std::find_if(s.begin(), s.end(), [](int ch)
+        {
+            return !std::isspace(ch);
+        }));
+        s.erase(std::find_if(s.rbegin(), s.rend(), [](int ch)
+        {
+            return !std::isspace(ch);
+        }).base(), s.end());
+    };
+
+    auto listFilesInDirectory = [](const std::string & path, std::vector<std::string>& out)
```

Comment:
Same. Maybe in the file `shared_modules/utils/fileSystem.h` or `shared_modules/utils/filesystemHelper.h` . Function `enumerateDir ` of `shared_modules/utils/filesystemHelper.h`?.

Reply:
Done!